src/misc/fs_utils.cpp

```diff
@@ -143,3 +144,21 @@ std::time_t to_time_t(const std_fs::file_time_type &fs_time)
 
 	return system_clock::to_time_t(sys_time);
 }
+
+// Searches the directory for the filename. Returns the first file that matches
+// the filename, regardless of casing.
+std::optional<std_fs::path> case_insensitive_search(const std_fs::path& dir,
+                                                    const std::string& filename)
+{
+	std::string filename_lower = filename;
+	lowcase(filename_lower);
+	std::error_code ec;
+	for (const auto& entry : std_fs::directory_iterator(dir, ec)) {
+		std::string itr_lower = entry.path().filename().string();
```

Comment:
The error code is set true if the iterator has a problem; suggest add a check and continue first thing in the for loop:

``` c++
if (ec) // Skip invalid or broken content {
    continue;
}
```

(all std::filesystem calls clear the error_code first thing, so we don't have to reset/clear it)

Reply:
Ah, OK.  I suppose I was mistaken in thinking the iterator could only error on construction rather than every iteration.  Added that check.

Reply:
indeed; that aspect isn't very intuitive (that it re-applies `ec` for each iteration).